src/server/services/cla.js

```diff
@@ -569,73 +575,76 @@ class ClaService {
+            'created_at': 1,
+            'repoId': 1,
+            'gist_url': 1,
+            'gist_version': 1,
+            'revoked_at': 1,
+        }, {
+            sort: {
+                'created_at': -1
+            }
+        })
 
-            clas.forEach((cla) => {
-                if (repoList.indexOf(cla.repo) < 0) {
-                    repoList.push(cla.repo)
-                    claList.push(cla)
-                }
-            })
-        }
-        try {
-            const repos = await repoService.all()
-            repos.forEach((repo) => {
-                selector.push({
-                    user: user.login,
-                    repo: repo.repo,
-                    gist_url: repo.gist,
-                    revoked_at: {$gt: date}
-                })
-                selector.push({
-                    user: user.login,
-                    repo: repo.repo,
-                    gist_url: repo.gist,
-                    revoked_at: undefined
-                })
-            })
-        } catch (error) {
-            logger.warn(new Error(error).stack)
-        }
+        // return only the ones which are not yet revoked
+        clas = clas.filter(cla => cla.revoked_at == undefined)
+        // return empty array if we don't have any repositories anymore
+        if (clas.length == 0) { return [] }
+
+        // find all repositories related to the clas and only return still existing repositories and where the signature is up-to-date
+        const repofilter = clas.map(cla => ({
+            owner: cla.owner,
+            gist: cla.gist_url
+        }))
+        const repos = await Repository.find({
+            $or: repofilter
+        })
 
-        const repoList = []
-        const uniqueClaList = []
-        try {
-            await findCla({
-                $or: selector
-            }, repoList, uniqueClaList)
-        } catch (error) {
-            logger.warn(new Error(error).stack)
-        }
-        try {
-            await findCla({
-                $or: [{
-                    user: user.login,
-                    revoked_at: {$gt: date}
-                },{
-                    user: user.login,
-                    revoked_at: undefined
-                }]
-            }, repoList, uniqueClaList)
-        } catch (error) {
-            logger.warn(new Error(error).stack)
+        // find all owners/organizations related to the clas and only return still existing owners and where the signature is up-to-date
+        const ownerfilter = clas.map(cla => ({
+            org: cla.owner,
+            gist: cla.gist_url
+        }))
+        const owners = await Org.find({
+            $or: ownerfilter
+        })
+
+        // filter if its an org CLA (repo == undefined) or if repo exists
+        clas = clas.filter(cla => {
+            // check if it is an owner or repo configuration
+            // if repo is undefined it is an org/owner wide cla
+            if (cla.repo == undefined) {
```

Comment:
```suggestion
            if (cla.org_cla) {
```
Just a suggestion, for two reasons:
 - readability
 - it would be nice (maybe someday in the future) to collect repo information (from which repo the cla was signed) in the orgwide clas as well. This was also requested from our users some time ago...

Reply:
changed in 6195c42a62c7b4e9fd6d1a4f076fc063766753e6